### ISSUE TITLE
assembling project does not add buildconfig package to jar

### ISSUE THREAD

**Issue by ghost** [mfuerstenau/gradle-buildconfig-plugin#5]
I have been using this plugin for running my tests and its great.  I have had to create a JAR file which is then used in a maven project. I am having issues with my maven build system saying package  `HelpersBuildConfig` does not exist.

`./gradlew assemble`
I unzipped my built jar file and realized that the compile `HelpersBuildConfig.java` is not part of the the JAR.

This issue might not be caused by this plugin. How do I go about resolving this? How can I add this generated java file into my jar?

**Comment by mfuerstenau** [comment]
The compiler inlines the constants. I could work aorund by making them available not as constants but by using getters. I'll try to include a configuration option to have getters rather than the direct access to the constants.

The problematic is sibject to this stackoverflow discussion: http://stackoverflow.com/questions/3524150/is-it-possible-to-disable-javacs-inlining-of-static-final-variables

**Comment by mfuerstenau** [comment]
Okay, Identified the other problem. The plugin generates the buildconfig class file and adds ot as a dependency but this dependency is not included in the jar by default. But a workaround would be 

``` groovy
afterEvaluate {
    jar {
        from tasks['compileBuildConfig'].outputs
    }
}
```

This way the class file is included in the jar. I could propbably make this the default.
Of course the task name is the default task name. It may differ if you changed the defaults or sourcesets.

**Comment by mfuerstenau** [comment]
Okay, next release will have the buildconfig added to sourceset classpath and sourceset output rather than added as a dependency. That way the build config class file will be included. I have tagged a SNAPSHOT that I sadly cannot upload to bintray. But you may clone and build it yourself to try. I still have to evaluate sife effects but I will release 1.1.5 soon.

**Comment by mfuerstenau** [comment]
@hariskhaliq 
You can try this test build, of course you have to remove the workaround from above.

``` gradle
buildscript {
   repositories {
      maven {
         url 'https://dl.bintray.com/mfuerstenau/maven/'
      }
   }
   dependencies {
      classpath 'gradle.plugin.de.fuerstenau:BuildConfigPlugin:1.1.5_beta_1'
   }
}
/* ... */
apply plugin: 'de.fuerstenau.buildconfig'
/* ... */
```

I will remove this test build once I release the official 1.1.5.

**Comment by ghost** [comment]
@mfuerstenau  Amazing. The workaround works great. 

I will just use the workaround until the actual release. 👍

**Comment by mfuerstenau** [comment]
I just release buildconfig plugin ~~1.1.5~~ (incomplete due to merge issues) 1.1.6 refactoring a lot and adding quite a few scenarios as tests. Feel free to update. I am closing this issue.

Please notice that I made a mistake merging 1.1.5, the commits merged are incomplete, and released version 1.1.6 immediately after to correct my error.
